### PR TITLE
Testing legacy code

### PR DESCRIPTION
**Pull request by s-furlong**

### CODE REVIEW

File: ruby/spec/gilded_rose_spec.rb [hunk 1]
```diff
@@ -9,12 +9,78 @@
...
+      expect(items[0].quality).to eq 31
+    end 
+ 
+    it "never allows quality to increase over 50" do
+      items = [Item.new("Aged Brie", 3, 50)]
+      GildedRose.new(items).update_quality()
+      expect(items[0].quality).to eq 50
+    end
+  
+    it "does not need to decrease qulaity for Sulfuras" do
```
Comment (breadoliveoilsalt) [inline ruby/spec/gilded_rose_spec.rb]: Spelling nit here in description.  Also, let's discuss using constants for strings to make your life easier.

File: ruby/spec/gilded_rose_spec.rb [hunk 2]
```diff
@@ -9,12 +9,78 @@
...
+      items = [Item.new("Sulfuras, Hand of Ragnaros", 3, 30)]
+      GildedRose.new(items).update_quality()
+      expect(items[0].quality).to eq 30
+    end
+   
+    it "does not need to sell Sulfuras" do
```
Comment (breadoliveoilsalt) [inline ruby/spec/gilded_rose_spec.rb]: Not sure how the test description matches the test...`does not decrease in quality of item is Sulfuras` perhaps?

File: ruby/spec/gilded_rose_spec.rb [hunk 3]
```diff
@@ -9,12 +9,78 @@
...
+      items = [Item.new("Sulfuras, Hand of Ragnaros", 0, 30)]
+      GildedRose.new(items).update_quality()
+      expect(items[0].quality).to eq 30
     end
     
+    it "allows back stage passes by one more than 10 days before the concert" do
```
Comment (breadoliveoilsalt) [inline ruby/spec/gilded_rose_spec.rb]: Missing some words here.  `to increase`?  Take a look at your test descriptions with fresh eyes in general, as long as you'll be making some changes. Or `it increases back stage passes`?

File: ruby/spec/gilded_rose_spec.rb [hunk 4]
```diff
@@ -9,12 +9,78 @@
...
+      items = [Item.new("Backstage passes to a TAFKAL80ETC concert",11, 30)]
+      GildedRose.new(items).update_quality()
+      expect(items[0].quality).to eq 31
+    end
+
+    it "allows the backstage passes to increase by 2 in quality 10 to 6 days before the concert" do
+      items = [Item.new("Backstage passes to a TAFKAL80ETC concert",9, 30)]
```
Comment (breadoliveoilsalt) [inline ruby/spec/gilded_rose_spec.rb]: Watch spacing before commas.

**Review by breadoliveoilsalt — COMMENTED** [review]
Quick comments (sorry, it's the end of the day) [Updated in the morning -- let's discuss live quickly this morning]
- overall very good - MUCH improved descriptions and great use of arrange/act/assert
- we should discuss a few things including
  - testing with multiple items
  - use constants, both in upper scope (eg, complicated names of products) and lower scope (eg, thing you are highlighting in the test)
  - breaking the test suite/file into two sections (or more) with more describe blocks -- non-special items vs special items.  I'm still chewing on this myself.  There are different possible variations I reckon.  For example...

 ```
describe GuildedRose do
  describe #update_quality do
    describe 'when handling standard items' do
    end
    describe 'when handling special [exceptional?] items' do
    end
  end
end

# or

describe GuildedRose do
  describe #update_quality do
    describe 'changes to quality' do
    end
    describe 'changes to expiration time' do
    end
   # etc
  end
end

# or even a combination of the two 🙃 . Although I recognize that is extreme.
```

File: ruby/spec/gilded_rose_spec.rb [hunk 5]
```diff
@@ -9,12 +9,78 @@
...
+    it "degrades quality over time by one if not at sell in date" do
+      items = [Item.new("foo", 10, 30)]
       GildedRose.new(items).update_quality()
-      expect(items[0].quality).to eq 49
+      expect(items[0].quality).to eq 29
+    end
+
+    it "decreases sell in date by one day" do
+      items = [Item.new("foo", 10, 30)]
+      GildedRose.new(items).update_quality()
```
Comment (breadoliveoilsalt) [inline ruby/spec/gilded_rose_spec.rb]: Ha, I forget my ruby -- do we need a `()` after update_quality if there are no args?

File: ruby/spec/gilded_rose_spec.rb [hunk 6]
```diff
@@ -9,12 +9,78 @@
...
+      expect(items[0].quality).to eq 0
+    end
+  
+    it "increases quality over time if name is Aged Brie" do
+      items = [Item.new("Aged Brie", 3, 30)]
+      GildedRose.new(items).update_quality()
+      expect(items[0].quality).to eq 31
+    end 
+ 
+    it "never allows quality to increase over 50" do
```
Comment (breadoliveoilsalt) [inline ruby/spec/gilded_rose_spec.rb]: Only applies to Aged Brie?

File: ruby/spec/gilded_rose_spec.rb (hunk 1 above)
Comment (s-furlong) [reply ruby/spec/gilded_rose_spec.rb]: corrected spelling error

File: ruby/spec/gilded_rose_spec.rb (hunk 2 above)
Comment (s-furlong) [reply ruby/spec/gilded_rose_spec.rb]: this test was a duplicate so it has been removed

File: ruby/spec/gilded_rose_spec.rb (hunk 6 above)
Comment (s-furlong) [reply ruby/spec/gilded_rose_spec.rb]: add test for backstage passes which is the other item that would increase in value

**Pull request merged**